### ISSUE TITLE
The one true master IBC implementation tracking issue

### ISSUE THREAD

**Issue by cwgoes** [cosmos/gaia#201]
*Step 1: _COMPLETED_ working demo on the `ibc-alpha` branch that can be played with externally.*

Blocked by:
- [x] Basic relayer implementation - https://github.com/cosmos/relayer (@jackzampolin)
  * [x] Working create and update client - https://github.com/cosmos/relayer/issues/27 (@cwgoes, @jackzampolin)
  * [x] Working handshakes and token transfers - https://github.com/cosmos/relayer/issues/21 (@cwgoes, @AdityaSripal, @jackzampolin)
  * [x] Implement IBC transactions - https://github.com/cosmos/relayer/issues/10 (@jackzampolin)
  * [x] Implement IBC queries - https://github.com/cosmos/relayer/issues/19 (@jackzampolin)
- [x] Proof debugging - https://github.com/cosmos/cosmos-sdk/pull/5697 (@cwgoes, @AdityaSripal)
- [x] Lite Client changes for relayer - https://github.com/tendermint/tendermint/pull/4479 (@melekes)

*Step 2: Spec-compatible 1.0 implementation for Game of Zones.*

Blocked by:
- [ ] Follow-ups from ADR 15 - https://github.com/cosmos/cosmos-sdk/issues/5507 (@fedekunze, @cwgoes)
- [ ] Migrate to `proto` for state storage & proofs - https://github.com/cosmos/cosmos-sdk/pull/5704 (@fedekunze)
- [ ] ICS 8 & standardisation on Merkle proofs - https://github.com/cosmos/ics/milestone/6 (@cwgoes) 
- [ ] Switch to using a prefix store for client impls - https://github.com/cosmos/cosmos-sdk/issues/5502 (blocked on above, @cwgoes & @fedekunze)
- [ ] Documentation on potential ICS 23 attack vectors - https://github.com/cosmos/cosmos-sdk/issues/5440 (@cwgoes)
- [ ] ADR 3 Dynamic Capabilities implementation - https://github.com/cosmos/cosmos-sdk/issues/5542  (@AdityaSripal)

Also see the SDK milestone for SDK issue tracking - https://github.com/cosmos/cosmos-sdk/milestone/21.

Also see all issues tagged `x/ibc` - https://github.com/cosmos/cosmos-sdk/issues?q=is%3Aissue+is%3Aopen+label%3Ax%2Fibc

Completed work:

- [x] ADR 3 spec - https://github.com/cosmos/cosmos-sdk/pull/5469
- [x] `lite2` integration - https://github.com/cosmos/relayer/pull/9 (@cmwaters, @jackzampolin, @melekes)
- [x] Tendermint client - https://github.com/cosmos/cosmos-sdk/pull/5485 (@fedekunze)
- [x] Set up CI to run demo in the `cosmos/relayer` repo
- [x] Revision of client type store semantics - https://github.com/cosmos/ics/pull/348 (@cwgoes)
- [x] ICS 4 tests - @fedekunze - https://github.com/cosmos/cosmos-sdk/pull/5578 (@jackzampolin)
- [x] ADR 4 spec - https://github.com/cosmos/cosmos-sdk/pull/5494
- [x] ADR 15 implementation - https://github.com/cosmos/cosmos-sdk/pull/5401
- [x] ADR 4 implementation - https://github.com/cosmos/cosmos-sdk/pull/5494 (@alexanderbez)
- [x] ADR 15 spec merged - https://github.com/cosmos/cosmos-sdk/pull/5230
- [x] Address ICS 2 misbehaviour handling - https://github.com/cosmos/cosmos-sdk/pull/5321
- [x] Historical header module (ADR / implementation) - https://github.com/cosmos/cosmos-sdk/pull/5340
- [x] Updated membership verification - https://github.com/cosmos/cosmos-sdk/pull/5335
- [x] Historical info support - https://github.com/cosmos/cosmos-sdk/pull/5475 (@fedekunze )
- [x] Fix account prefix issue - https://github.com/cosmos/cosmos-sdk/pull/5581 (@alessio)
- [x] ICS `20-transfer` 
  * [x] Update spec to reflect changes caused by ADR 15
  * [x] Add tests for receiving packets & handling timeouts @fedekunze 
  * [x] Queriers and CLI - https://github.com/cosmos/cosmos-sdk/issues/5490 @fedekunze 
- [x] Update Gaia to IBC `ibc-alpha` code: https://github.com/cosmos/gaia/pull/288
- [x] ICS 7 follow-ups - https://github.com/cosmos/cosmos-sdk/pull/5666 (@AdityaSripal)

Then we need an updated demo with Gaia in this repository.

**Comment by cwgoes** [comment]
@fedekunze Is this still up-to-date?

edit: I have updated it (as best I understand).

**Comment by cwgoes** [comment]
Closing in favour of the [SDK milestone](https://github.com/cosmos/cosmos-sdk/milestone/21).
